src/rgw/rgw_file.cc

```diff
@@ -844,11 +862,15 @@ namespace rgw {
     return rc;
   } /* RGWFileHandle::write */
 
-  int RGWFileHandle::close()
+  int RGWFileHandle::write_finish(uint32_t flags)
   {
-    lock_guard guard(mtx);
-
+    unique_lock guard{mtx, std::defer_lock};
     int rc = 0;
+
+    if (! (flags & FLAG_LOCKED)) {
+      guard.lock();
+    }
+
```

Comment:
This is bad style. A unique_lock is the lock and flag all in one, if the function expects to be called with the lock both locked and not locked, it should take a unique_lock by reference.